packages/clay-dataset-display/src/ClayDatasetDisplay.js

```diff
@@ -0,0 +1,472 @@
+	 * list.
+	 * @param {!Event} event
+	 * @private
+	 */
+	handleItemToggled_(event) {
+		let checkedStatus = event.target.checked;
+		let itemId = event.target.getAttribute('value');
+
+		if (!checkedStatus) {
+			for (let [index, item] of this.selectedItems_.entries()) {
+				if (
+					item[
+						this.views[this.selectedView].schema.inputValueField
+					] === itemId
+				) {
+					item.selected = false;
+					this.selectedItems_.splice(index, 1);
+					break;
+				}
+			}
+		} else {
+			let found = false;
+
+			if (this.items) {
+				for (let item of this.items) {
+					if (item.items) {
+						for (let childrenItem of item.items) {
+							if (
+								childrenItem[
+									this.views[this.selectedView].schema
+										.inputValueField
+								] === itemId
+							) {
+								childrenItem.selected = true;
+								this.selectedItems_.push(childrenItem);
+								found = true;
+								break;
+							}
+						}
+					} else {
+						if (
+							item[
+								this.views[this.selectedView].schema
+									.inputValueField
+							] === itemId
+						) {
+							item.selected = true;
+							this.selectedItems_.push(item);
+							found = true;
+						}
+					}
+
+					if (found) {
+						break;
+					}
+				}
+			}
+		}
+
+		this.items = this.items;
+		this.selectedItems_ = this.selectedItems_;
+	}
+
+	/**
+	 * Continues the propagation of the plus button clicked event
+	 * @param {!Event} event
+	 * @private
+	 */
+	handlePlusButtonClicked_(event) {
+		this.emit('plusButtonClicked', event);
+	}
+
+	/**
+	 * Continues the propagation of the search event
+	 * @param {!Event} event
+	 * @private
+	 */
+	handleSearch_(event) {
+		this.emit('search', event);
+	}
+
+	/**
+	 * Continues the propagation of the sorting button clicked event
+	 * @param {!Event} event
+	 * @private
+	 */
+	handleSortingButtonClicked_(event) {
+		this.emit('sortingButtonClicked', event);
+	}
+
+	/**
+	 * Selects all items on management toolbar select all button click.
+	 * @private
+	 */
+	handleSelectAllClicked_() {
+		this.selectAllItems();
+	}
+
+	/**
+	 * Changes the selection status of the dataset items on management toolbar
+	 * select page checkbox change.
+	 * @param {!Event} event
+	 * @private
+	 */
+	handleSelectPageCheckboxChanged_(event) {
+		let checkboxStatus = event.target.checked;
+
+		if (checkboxStatus) {
+			this.selectAllItems();
+		} else {
+			this.deselectAllItems();
+		}
+	}
+
+	/**
+	 * Changes the view on management toolbar view type click.
+	 * @param {!Event} event
+	 * @private
+	 */
+	handleViewTypeClicked_(event) {
+		this.views[this.selectedView].active = false;
+
+		for (let [index, view] of this.views.entries()) {
+			if (view === event.viewType) {
+				this.views[index].active = true;
+				this.selectedView = index;
+				break;
+			}
+		}
+	}
+
+	/**
+	 * Iterates over the selected items array, mark all as not selected and
+	 * removes them from the selectedItems array.
+	 * @private
+	 */
+	deselectAllItems() {
```

Comment:
Would not it be better to follow the default `deselectAllItems_`? Or was it intentional?

Reply:
D'oh!